scipy/optimize/_trustregion_exact.py

```diff
@@ -0,0 +1,424 @@
+"""Nearly exact trust-region optimization subproblem."""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.linalg import norm, get_lapack_funcs, solve_triangular, norm
+from ._trustregion import (_minimize_trust_region, BaseQuadraticSubproblem)
+
+__all__ = ['estimate_smallest_singular_value',
+           'singular_leading_submatrix',
+           'IterativeSubproblem']
+
+
+def _minimize_trustregion_exact(fun, x0, args=(), jac=None, hess=None,
+                                **trust_region_options):
+    """
+    Minimization of scalar function of one or more variables using
+    a nearly exact trust-region algorithm.
+
+    Options
+    -------
+    initial_trust_radius : float
+        Initial trust-region radius.
+    max_trust_radius : float
+        Maximum value of the trust-region radius. No steps that are longer
+        than this value will be proposed.
+    eta : float
+        Trust region related acceptance stringency for proposed steps.
+    gtol : float
+        Gradient norm must be less than ``gtol` before successful
+        termination.
+    """
+
+    if jac is None:
+        raise ValueError('Jacobian is required for dogleg minimization')
+    if hess is None:
+        raise ValueError('Hessian is required for dogleg minimization')
+    return _minimize_trust_region(fun, x0, args=args, jac=jac, hess=hess,
+                                  subproblem=IterativeSubproblem,
+                                  **trust_region_options)
+
+
+def estimate_smallest_singular_value(c):
```

Comment:
> It estimate smin the smallest singular value (not the condition number!) of the triangular matrix C.
I will try to make this more clear at the comments.

I think it's clear from the documentation, I just didn't really pay attention.

> This algorithm is actually hard to grasp, and most references talk about it in the context of condition number,

I think I've figured it. The procedure is explained in section 3 of [1]_ with `Q` being identity (perhaps worth including this line in the code).  Also maybe it's better to remove the second reference, because while it has a useful algorithm listing the discussion is a bit misleading for our problem.